File: benchmark/c/cpw/check_number_power_k_using_base_changing_method/check_number_power_k_using_base_changing_method.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int check_number_power_k_using_base_changing_method ( unsigned int n, unsigned int k ) ;
int check_number_power_k_using_base_changing_method ( unsigned int n, unsigned int k ) {
  bool oneSeen = 0;
  while ( n > 0 ) {
    int digit = n % k;
    if ( digit > 1 ) return 0;
    if ( digit == 1 ) {
      if ( oneSeen ) return 0;
      oneSeen = 1;
    }
    n /= k;
  }
  return 1;
}

File: translation_result/else_bms/fail/c/cpw/check_number_power_k_using_base_changing_method/gpt4/BaseRepair/check_number_power_k_using_base_changing_method.rs

fn check_number_power_k_using_base_changing_method(n: u32, k: u32) -> u32 {
    let mut one_seen = false;
    let mut n = n;

    while n > 0 {
        let digit = n % k;
        if digit > 1 {
            return 0;
        }
        if digit == 1 {
            if one_seen {
                return 0;
            }
            one_seen = true;
        }
        n /= k;
    }
    return 1;
}
